app/components/emoji_picker/emoji_picker.js

```diff
@@ -0,0 +1,249 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {injectIntl, intlShape} from 'react-intl';
+import {
+    Dimensions,
+    SectionList,
+    StyleSheet,
+    Text,
+    TouchableOpacity,
+    View
+} from 'react-native';
+
+import Emoji from 'app/components/emoji';
+import SearchBar from 'app/components/search_bar';
+import {emptyFunction} from 'app/utils/general';
+import {makeStyleSheetFromTheme, changeOpacity} from 'app/utils/theme';
+
+const {width: deviceWidth} = Dimensions.get('window');
+const EMOJI_SIZE = 30;
+const EMOJI_GUTTER = 7.5;
+const SECTION_MARGIN = 15;
+
+class EmojiPicker extends PureComponent {
+    static propTypes = {
+        emojis: PropTypes.array.isRequired,
+        intl: intlShape.isRequired,
+        onEmojiPress: PropTypes.func,
+        theme: PropTypes.object.isRequired
+    };
+
+    static defaultProps = {
+        onEmojiPress: emptyFunction
+    };
+
+    leftButton = {
+        id: 'close-edit-post'
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            emojis: props.emojis,
+            searchTerm: ''
+        };
+    }
+
+    changeSearchTerm = (text) => this.setState({searchTerm: text})
+
+    cancelSearch = () => this.setState({searchTerm: ''})
+
+    buildData = () => {
+        const {emojis, searchTerm} = this.state;
+
+        const nextEmojis = emojis.map((section) => {
+            const {title, data} = section;
+
+            const nextData = data.reduce((s, emoji) => {
```

Comment:
Yeah I think you're right. This is definitely leftover from trying to make up for the flexbox layout issue. Will make the change.